BirtherDay/Common/Manager/BDNavigationPathManager.swift

```diff
@@ -0,0 +1,55 @@
+//
+//  BDNavigationPathManager.swift
+//  BirtherDay
+//
+//  Created by Rama on 5/29/25.
+//
+
+import SwiftUI
+
+/// 홈뷰로부터 가능 Path 종류
+enum BDAppPath: Hashable {
+    case create(BDCreateCouponPath)
+    case myCoupon(BDMyCouponPath)
+}
+
+enum BDCreateCouponPath: Hashable, Equatable {
+    case selectTemplate
+    case couponInfo
+    case couponLetter
+    case couponPicture
+    case couponComplete
+}
+
+enum BDMyCouponPath: Hashable, Equatable {
+    case couponInventory
+    case couponDetail
+    case interaction
+    case interactionComplete
+}
+
+class BDNavigationPathManager: ObservableObject {
+    /// Path Stack
+    @Published var appPaths: [BDAppPath] = []
+    
+    /// Ceate Path에서 뷰 전환
+    func addCreatePath(_ path: BDCreateCouponPath) {
+        appPaths.append(.create(path))
+    }
+    
+    /// MyCoupon Path에서 뷰 전환
+    func addMyCouponPath(_ path: BDMyCouponPath) {
+        appPaths.append(.myCoupon(path))
+    }
+    
+    /// 홈뷰로 가기
+    func removeAllPath() {
+        appPaths.removeAll()
+    }
+    
+    /// 뒤로 가기
+    func popPath() {
+        appPaths.popLast()
+    }
```

Comment:
method 명에서 동사를 add, remove, pop으로 사용했는데 stack이다 보니

push, pop으로 사용하는 것이 좋을 것 같습니다.

그리고 removeAll 보다는 clear가 더 짧고 명확한 표현같아용!

```suggestion
       /// BDCreateCouponPath를 네비게이션 스택에 추가
    func pushCreatePath(_ path: BDCreateCouponPath) {
        appPaths.append(.create(path))
    }

    /// BDMyCouponPath를 네비게이션 스택에 추가
    func pushMyCouponPath(_ path: BDMyCouponPath) {
        appPaths.append(.myCoupon(path))
    }

    /// 모든 경로 제거 (홈으로 이동)
    func clearPaths() {
        appPaths.removeAll()
    }

    /// 마지막 경로 제거 (뒤로가기)
    func popPath() {
        appPaths.popLast()
    }
```

Reply:
혹은 **모든 경로 제거** 함수를
```swift
func goToRoot()
```
도 괜찮아보이네요!

Reply:
반영하겠습니다 ^_^